pkg/kubelet/dockertools/manager.go

```diff
@@ -1401,6 +1401,12 @@ func (dm *DockerManager) computePodContainerChanges(pod *api.Pod, runningPod kub
 			containersToKeep[containerID] = index
 			continue
 		}
+
+		// Record an event for the healthcheck failure.
+		ref, err := kubecontainer.GenerateContainerRef(pod, &container)
+		if err == nil {
+			dm.recorder.Eventf(ref, "failed", "Healthcheck failed with probe result: %v", result)
+		}
```

Comment:
Ah good call, I forgot about @thockin's PR. We should definitely merge both of these (more likely use his which I think subsumes this one)